src/main/java/fact/io/hdureader/ZFITSHeapReader.java

```diff
@@ -130,17 +132,56 @@ public static ZFITSHeapReader forTable(BinTable binTable){
      *
      * The old ZFits implementation by M.Bulinski ignored the problem by default.
      *
-     * @param ignoreWrongTileHeader if true, ognores wrong tileheader information.
+     * @param ignoreWrongTileHeader if true, ignores wrong tileheader information.
      * @return a map containing the the data from the rows columns.
      * @throws NoSuchElementException iff hasNext() is false
      */
     public OptionalTypesMap<String, Serializable> getNextRow(boolean ignoreWrongTileHeader) throws IOException {
         if(!hasNext()){
            throw new NoSuchElementException();
         }
+        // check if we currently still have rows in the cache if not get the next tile
+        if (maxRowsInCache==curRowInCache) {
```

Comment:
Since this iterator only walks forward a simple queue might suffice here right? No need to bookkeep three variables if you can use a datastructure that does the bookmarking automatically.

Reply:
You would need a countdown type class, a queue would be a little bit of an overkill but yeah would work. Also, for this we only use two variables, so not much to gain anyways.